be1-go/channel/lao/mod.go

```diff
@@ -675,10 +579,15 @@ func (c *Channel) processRollCallOpen(msg message.Message, action string) error
 }
 
 // processRollCallClose processes a close roll call message.
-func (c *Channel) processRollCallClose(msg messagedata.RollCallClose, socket socket.Socket) error {
+func (c *Channel) processRollCallClose(msg message.Message, msgData interface{}, senderSocket socket.Socket) error {
+
+	data, ok := msgData.(*messagedata.RollCallClose)
+	if !ok {
+		return xerrors.Errorf("message %v isn't a rollcall#close message", msgData)
+	}
 
 	// check that data is correct
-	err := c.verifyMessageRollCallClose(msg)
+	err := c.verifyMessageRollCallClose(*data)
```

Comment:
you can update the verify functions to take pointers. At this point it doesn't make sense to create a copy.